pkg/daemon/drain.go

```diff
@@ -39,12 +39,21 @@ func (dn *Daemon) cordonOrUncordonNode(desired bool) error {
 	}
 	var lastErr error
 	if err := wait.ExponentialBackoff(backoff, func() (bool, error) {
+		// Log has been added to ensure that MCO is correctly performing cordon/uncordon.
+		// This should help us with debugging bugs like https://bugzilla.redhat.com/show_bug.cgi?id=2022387
+		glog.Infof("Initiating %s on node (currently schedulable: %t)", verb, !dn.node.Spec.Unschedulable)
 		err := drain.RunCordonOrUncordon(dn.drainer, dn.node, desired)
 		if err != nil {
 			lastErr = err
 			glog.Infof("%s failed with: %v, retrying", verb, err)
 			return false, nil
 		}
+		if dn.node.Spec.Unschedulable != desired {
```

Comment:
I think we should do a get of the node from the lister or from the API to make sure the API node reflects reality.

Reply:
good idea, fixed.